### DISCUSSION TITLE
Copy icon button other uses

### DISCUSSION THREAD

**Discussion by Fazle-Jahangir** [twilio-labs/paste#3383] [General] [answered]
I am currently adding icon only buttons inside a data grid component. They are: 

1. Edit
2. Duplicate
3. Delete

There was a misunderstanding where I used the "copy" icon from the library to duplicate a row with the same content. Below is a screenshot of the data grid with the action buttons in the last column.

**Question**: Do you recommend using this icon for duplication or are there strict guidelines to not use icons outside of their intended purpose, in this case the icon is only meant to copy text. If no then I plan to use a secondary "duplicate" button to replace the icon button. Perhaps for the future would it be possible to add a new duplicate icon?

<img width="724" alt="Screenshot 2023-08-04 at 13 54 38" src="https://github.com/twilio-labs/paste/assets/100706574/96764981-321a-41b8-948f-31b2e73592b6">

**Comment by serifluous** [comment]
Hey @Fazle-Jahangir ! By "misunderstanding", do you mean with your team members or with customers?

Cases like this are why we recommend you pair text with icons so the action can be clearer, even if you use the Copy icon as a proxy for a Duplicate action. If you've tested this with customers, and the icon causes confusion, please submit a request to our dependency intake form for a Duplicate icon: https://airtable.com/appwvsQKyJHfnG5Sa/shrvMvL3BG6Zm0f9l

Thanks!

**Reply by Fazle-Jahangir** [reply]
Hi @serifluous I meant with team members, this hasn't been shipped to customers yet. I'll look into adding text. Do you recommend or is it perhaps acceptable using this icon as a proxy for the action "duplicate" ?

**Reply by serifluous** [reply, marked as answer]
When we add new icons, we want to consider a few things:

1. Is this a one-off use case, where an existing icon would be understandable enough to customers? If we were ever to change the glyph of the existing icon, would that completely change the meaning in the one-off use case?
2. Does the maintenance/support cost of one additional icon outweigh the customer benefit? Does the customer risk of using the icon as a proxy outweigh the maintenance/support cost?

So I'd really say, it's kind of up to you to tell us 🙃 . Are there more cases where a Duplicate action would appear? Would you be able to find out if customers find the copy icon sufficiently understandable in the Duplicate action?

Either way, we're not opposed to adding a Duplicate icon, nor do we have _very_ strict guidelines about using icons for closely related one-off actions. It's just hard for us to make a decision either way with limited information! One thing I'd suggest is a quick message in #team-ux about whether other folks might have a Duplicate action in their part of the product.

**Reply by Fazle-Jahangir** [reply]
Posted in team-ux for more visibility, good suggestion. 

I am confident that the duplicate action is not used elsewhere in the Flex console as far as I am aware. This icon when used is usually within a read-only combo-box as a suffix where the user is able to copy the text in the field. So if we use it as an icon-only button with perhaps a label or a tooltip appears on hover for example, it should differentiate itself from the copy action in read-only text fields.
